src/server/routes/queueDelete.ts

```diff
@@ -0,0 +1,174 @@
+import { Request, Response } from 'express';
+import { Router } from 'express';
+import { checkSchema, Schema } from 'express-validator';
+import { readClient } from '../../database/client';
+import { SavedItemDataService } from '../../dataService';
+import config from '../../config';
+import { sqs } from '../../aws/sqs';
+import {
+  SendMessageBatchCommand,
+  SendMessageBatchCommandOutput,
+  SendMessageBatchRequestEntry,
+} from '@aws-sdk/client-sqs';
+import { nanoid } from 'nanoid';
+import * as Sentry from '@sentry/node';
+import { validate } from './validator';
+
+export type SqsMessage = {
+  userId: number;
+  email: string;
+  status: 'FREE' | 'PREMIUM';
+  itemIds: number[];
+};
+
+const router = Router();
+
+const batchDeleteSchema: Schema = {
+  traceId: {
+    in: ['body'],
+    optional: true,
+    isString: true,
+    notEmpty: true,
+  },
+  userId: {
+    in: ['body'],
+    errorMessage: 'Must provide valid userId',
+    isInt: true,
+    toInt: true,
+  },
+  email: {
+    in: ['body'],
+    errorMessage: 'Must provide valid email',
+    isEmail: true,
+  },
+  status: {
+    in: ['body'],
+    errorMessage: 'Must provide valid status (FREE or PREMIUM)',
+    custom: {
+      options: (status) => ['FREE', 'PREMIUM'].includes(status),
+    },
+  },
+};
+
+router.post(
+  '/',
+  checkSchema(batchDeleteSchema),
+  validate,
+  (req: Request, res: Response) => {
+    const traceId = req.body.traceId ?? nanoid();
+    const savedItemService = new SavedItemDataService({
+      userId: req.body.userId.toString(),
+      dbClient: readClient(),
+      apiId: 'backend',
+    });
+
+    enqueueSavedItemIds(req.body, savedItemService, traceId);
+
+    return res.send({
+      status: 'OK',
+      message: `QueueDelete: Enqueued items for User ID: ${req.body.userId} (traceId='${traceId}')`,
+    });
+  }
+);
+
+/**
+ * Enqueue item IDs for deletions in batches
+ * @param data
+ * @param savedItemService
+ * @param traceId
+ */
+export async function enqueueSavedItemIds(
+  data: Omit<SqsMessage, 'itemIds'>,
+  savedItemService: SavedItemDataService,
+  traceId: string
+): Promise<void> {
+  const { userId, email, status } = data;
+  const limit = config.queueDelete.queryLimit;
+  let offset = 0;
+  const sqsSends: Promise<SendMessageBatchCommandOutput>[] = [];
+  let sqsEntries: SendMessageBatchRequestEntry[] = [];
+
+  const loopCondition = true;
```

Comment:
this never gets updated

Reply:
ESlint yells at me when I just put `true` in the while loop